source/PlayerInfo.cpp

```diff
@@ -2564,76 +2519,151 @@ void PlayerInfo::ValidateLoad()
+	});
 
-	// Store conditions for flagship current crew, required crew, and bunks.
-	if(flagship)
+	// The number of active ships the player has of the given category
+	// (e.g. Heavy Warships).
+	auto &&shipTypesProvider = conditions.GetProviderPrefixed("ships: ");
+	shipTypesProvider.SetGetFun([this](const string &name) -> int64_t
 	{
-		conditions["flagship crew"] = flagship->Crew();
-		conditions["flagship required crew"] = flagship->RequiredCrew();
-		conditions["flagship bunks"] = flagship->Attributes().Get("bunks");
-		if(flagship->GetSystem())
-			conditions["flagship system: " + flagship->GetSystem()->Name()] = 1;
-		if(flagship->GetPlanet())
-			conditions["flagship planet: " + flagship->GetPlanet()->TrueName()] = 1;
-	}
-	else
+		int64_t retVal = 0;
+		for(const shared_ptr<Ship> &ship : ships)
+			if(!ship->IsParked() && !ship->IsDisabled() && ship->GetSystem() == system
+					&& name == "ships: " + ship->Attributes().Category())
+				++retVal;
+		return retVal;
+	});
+
+	// Conditions to determine if flagship is in a system and on a planet.
+	auto &&flagshipSystemProvider = conditions.GetProviderPrefixed("flagship system: ");
+	auto flagshipSystemFun = [this](const string &name) -> bool
 	{
-		conditions["flagship crew"] = 0;
-		conditions["flagship required crew"] = 0;
-		conditions["flagship bunks"] = 0;
-	}
+		if(!flagship || !flagship->GetSystem())
+			return false;
+		return name == "flagship system: " + flagship->GetSystem()->Name();
+	};
+	flagshipSystemProvider.SetHasFun(flagshipSystemFun);
+	flagshipSystemProvider.SetGetFun(flagshipSystemFun);
 
-	// Conditions for your fleet's attractiveness to pirates:
-	pair<double, double> factors = RaidFleetFactors();
-	conditions["cargo attractiveness"] = factors.first;
-	conditions["armament deterrence"] = factors.second;
-	conditions["pirate attraction"] = factors.first - factors.second;
+	auto &&flagshipPlanetProvider = conditions.GetProviderPrefixed("flagship planet: ");
+	auto flagshipPlanetFun = [this](const string &name) -> bool
+	{
+		if(!flagship || !flagship->GetPlanet())
+			return false;
+		return name == "flagship planet: " + flagship->GetPlanet()->TrueName();
+	};
+	flagshipPlanetProvider.SetHasFun(flagshipPlanetFun);
+	flagshipPlanetProvider.SetGetFun(flagshipPlanetFun);
+
+	// Read/write government reputation conditions.
+	// The erase function is still default (since we cannot erase government conditions).
+	auto &&reputationProvider = conditions.GetProviderPrefixed("reputation: ");
+	reputationProvider.SetHasFun([](const string &name) -> bool
+	{
+		string govName = name.substr(strlen("reputation: "));
+		return GameData::Governments().Has(govName);
+	});
+	reputationProvider.SetGetFun([](const string &name) -> int64_t
+	{
+		string govName = name.substr(strlen("reputation: "));
+		auto gov = GameData::Governments().Get(govName);
+		if(!gov)
+			return 0;
+		return gov->Reputation();
+	});
+	reputationProvider.SetSetFun([](const string &name, int64_t value) -> bool
```

Comment:
Just wondering, why is this the only one that has a setter?
reputation is the only primary condition, that is not generated by code and is written in the save file, right?

maybe it could be explained a comment

Reply:
Reputations with governments are stored in `GameData::Governments()` and they typically get copied between the conditions and the governments. This PR removes all the 2-directional copying and just performs reads and writes of those conditions directly in the `GameData::Governments()` reputation data.
This also means that reputations are not primary conditions (according to terminology in this PR they are derived conditions), and the reputation conditions will no longer be stored in savefiles, but the Government reputations still are.